File: Cryptography/python/monoalphabetic_cipher/testcases/test_monoalphabetic.py
#!/usr/bin/python

from monoalphabetic_cipher import *

def test_enc_dec():
	letters = range(ord('a'), ord('z')+1)
	letters = map(chr, letters)

	print 'Plaintext: ',
	plaintext = raw_input()

	print 'Keystring: ',
	keymap = raw_input().lower()

	'''
	Create a keymap by using the letters in the key provided and pad it by 
	appending the rest of the alphabet that aren't in the key
	'''
	for k in keymap:
		letters.remove(k)
	
	for l in letters:
		keymap += l

	print 'Keymap: ', keymap

	m = MonoalphabeticCipher(keymap)
	ciphertext = m.encrypt(plaintext)
	dec_plaintext = m.decrypt(ciphertext)

	print 'Ciphertext: {0}'.format(ciphertext)
	print 'Decrypted plaintext: {0}'.format(dec_plaintext)


if __name__ == "__main__":
	test_enc_dec()

File: Cryptography/haskell/monoalphabetic_cipher/testcases/test_monoalphabetic.hs
import MonoalphabeticCipher
import Char

main = do
	plaintext <- (getLine::IO String)
	keystring <- (getLine::IO String)

	let keystring_u = map toUpper keystring
	
	let keymap = keystring_u ++ [x | x <- ['A' .. 'Z'], notElem x keystring_u]
	let enc_key_map = generate_enc_key_map keymap
	let dec_key_map = generate_dec_key_map keymap
	
	putStrLn ("Enc Map: " ++ enc_key_map)
	putStrLn ("Dec Map: " ++ dec_key_map)

	let ciphertext = monoalphabetic_encrypt plaintext enc_key_map
	let dec_plaintext = monoalphabetic_decrypt ciphertext dec_key_map

	putStrLn ("Encrypted Ciphertext: " ++ ciphertext)
	putStrLn ("Decrypted Plaintext: " ++ dec_plaintext)
 
